sdks/python/apache_beam/typehints/trivial_inference.py

```diff
@@ -46,9 +48,12 @@ def instance_to_type(o):
   if o is None:
     return type(None)
   elif t not in typehints.DISALLOWED_PRIMITIVE_TYPES:
-    if t == types.InstanceType:
-      return o.__class__
-    elif t == BoundMethod:
+    try:
+      if t == types.InstanceType: # For old-style classes
```

Comment:
<!--new_thread; commit:b015963f671ceb39f8033014eb6d4e46a4a734bc; resolved:0-->
Why do not we guard this with `sys.version_info[0] == 2:` instead? We use that in other places.

Reply:
Yeah I'll need to change this.